File: 백준/Silver/10844. 쉬운 계단 수/쉬운 계단 수.java
    import java.util.*;
    import java.io.*;
    public class Main {

    	public static void main(String[] args) throws IOException {
    	BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
    	BufferedWriter bw=new BufferedWriter(new OutputStreamWriter(System.out));
    	int num=Integer.parseInt(br.readLine());
    	long arr[][]=new long[num][10];
    	for(int i=1;i<10;i++) {
    		arr[0][i]=1;
    	}
    	for(int j=1;j<num;j++) {
    		for(int k=0;k<10;k++) {
    			if(k==0)
    				arr[j][k]=arr[j-1][k+1]%1000000000;
    			else if(k==9)
    				arr[j][k]=arr[j-1][k-1]%1000000000;
    			else
    			arr[j][k]=arr[j-1][k-1]+arr[j-1][k+1]%1000000000;
    		}
    	}
    	long sum=0;
    	for(int k=0;k<10;k++) {
    		sum+=arr[num-1][k];
    	}
    	System.out.print(sum%1000000000);
}
    	}
    

File: 백준/Silver/10844. 쉬운 계단 수/쉬운 계단 수.py
n = int(input())
arr = [[0]*10 for _ in range(n)]
for i in range(1,10):
    arr[0][i] = 1
for i in range(1,n):
    for j in range(10):
        if j == 0:
            arr[i][j] = arr[i-1][j+1]%1000000000
        elif j == 9:
            arr[i][j] = arr[i-1][j-1]%1000000000
        else:
            arr[i][j] = arr[i-1][j-1] + arr[i-1][j+1]%1000000000

sum = 0
for i in range(10):
    sum += arr[n-1][i]
print(sum%1000000000)